src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -201,38 +200,55 @@ public sealed class PackageIndex : IPackageIndex {
             }
         }
 
+        private async Task UpdateInstalledPackagesAsync() {
+            var token = await _buildIndexLock.WaitAsync();
```

Comment:
- `token` isn't released, so there can be a deadlock
- I think there should be `ResetAsync` instead of `WaitAsync`